dataformat-aml/src/test/java/io/adminshell/aas/v3/dataformat/aml/serialize/AmlSerializerTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021 Fraunhofer-Gesellschaft zur Foerderung der angewandten Forschung e. V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.adminshell.aas.v3.dataformat.aml.serialize;
+
+import io.adminshell.aas.v3.dataformat.aml.fixtures.FullExample;
+import io.adminshell.aas.v3.dataformat.SerializationException;
+import io.adminshell.aas.v3.dataformat.aml.AmlSerializationConfig;
+import io.adminshell.aas.v3.dataformat.aml.AmlSerializer;
+import io.adminshell.aas.v3.dataformat.aml.serialization.id.IntegerIdGenerator;
+import io.adminshell.aas.v3.model.AssetAdministrationShellEnvironment;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Iterator;
+import org.junit.Test;
+import org.xml.sax.SAXException;
+import org.xmlunit.builder.DiffBuilder;
+import org.xmlunit.diff.DefaultNodeMatcher;
+import org.xmlunit.diff.Diff;
+import org.xmlunit.diff.Difference;
+import org.xmlunit.diff.ElementSelectors;
+
+public class AmlSerializerTest {
+
+    @Test
+    public void testSAPFullExample() throws SerializationException, SAXException, IOException {
+        validateAmlSerializer(FullExample.FILE, FullExample.ENVIRONMENT);
+    }
+
+    private void validateAmlSerializer(File expectedFile, AssetAdministrationShellEnvironment environment)
+            throws SerializationException, SAXException, IOException {
+        String expected = Files.readString(expectedFile.toPath());
+        String actual = new AmlSerializer().write(environment, AmlSerializationConfig.builder()
+                .idGenerator(new IntegerIdGenerator())
+                .build());
+        System.out.println(actual);
+        Diff diff = DiffBuilder
+                .compare(expected)
+                .withTest(actual)
+                .normalizeWhitespace()
+                .withNodeMatcher(new DefaultNodeMatcher(
+                        ElementSelectors.conditionalBuilder()
+                                .whenElementIsNamed("Attribute")
```

Comment:
These are instructions on how to compare the generated XML/AML to the expected XML/AML. The used framework only works on XML and compares nodes of the XML tree based on node name which in case of AML is most of the time either Attribute or InternalElement. As in AML each element typically has multiple sub-elements of type Attribute and/or InternalElement. These statements specify that e.g. an element is named "Attribute" it should be compare not with another element named "Attribute" at the same position in the other file but with another element that is also named "Attribute" and has the same value for attribute "Name". 
Basically, these rules define how to uniquely identify an Attribute/InternalElement/ExternalInterface so that order of elements is not important when comparing AML files.